aten/src/ATen/core/type.cpp

```diff
@@ -260,10 +271,13 @@ MatchTypeReturn matchTypeVariables(
       }
       return MatchTypeReturn::Success();
     } else if (auto tup_type = actual->cast<TupleType>()) {
-      auto maybe_tuple_unified = unifyTypeList(tup_type->elements());
+      std::stringstream ss;
+      auto maybe_tuple_unified = unifyTypeList(tup_type->elements(), ss);
       if (maybe_tuple_unified) {
         return matchTypeVariables(
             lt_formal->getElementType(), *maybe_tuple_unified, type_env);
+      } else {
```

Comment:
Could you remove this return or include more context to the error message here? 
I'm not sure `(Could not unify the type list element ...)` is a better error message than 

`    ss << "Cannot match " << lt_formal->python_str() << " to "
       << actual->python_str();`